content/tutorials/trainapp-fitbit.textile

```diff
@@ -0,0 +1,550 @@
+            action="Add Item"
+            onAutocomplete={value => {
+              const autoValues = JSON.parse(
+                props.settingsStorage.getItem("stationspossible")
+              ).values;
+              return autoValues.filter(option =>
+                option.name.toLowerCase().startsWith(value.toLowerCase())
+              );
+            }}
+          />
+        )}
+        {/* Getting possible towards depending on the line & station selected by the user, 
+        through the settingsStorage set in the companion */}
+        {props.settingsStorage.getItem("via") && (
+          <TextInput
+            title="Select Station Name"
+            label="Towards"
+            placeholder="Search station"
+            settingsKey="towards" // Used to read the selected value in companion
+            action="Add Item"
+            onAutocomplete={value => {
+              const autoValues = JSON.parse(
+                props.settingsStorage.getItem("via")
+              ).values;
+              return autoValues.filter(option =>
+                option.name.toLowerCase().startsWith(value.toLowerCase())
+              );
+            }}
+          />
+        )}
+      </Section>
+    </Page>
+  );
+}
+
+registerSettingsPage(train);
+```
+You can refer to the Settings guide "here":https://dev.fitbit.com/build/guides/settings/ .
+
+"See this step in Github":https://github.com/ably/tutorials/commit/a1c324a5f28b250ad11973f24298c1acefecd846
+
+h2(#train-data). Step 4 -  Adding static trains data
+
+We'll get started with a few import statements, add the following to your @index.js@ file which is inside the @companion@ folder:
+
+```[javascript]
+import Ably from "ably/browser/static/ably-commonjs.js";
+// Import the messaging module
+import * as messaging from "messaging";
+import { settingsStorage } from "settings";
+import { geolocation } from "geolocation";
+
+let navigator = {userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_2) 
+AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36"}
+```
+Then lets add the static data below it from the "gist":https://gist.github.com/vardhanapoorv/b96c5138e7a12b6e6b8f7a02d4d4e76a .
+It contains two arrays, one is used for @StationsIDMapping@ (line & station combination - @stationid@) and other is used for @towardsMapping@ (line & origin combination to get @towards@ options) . 
+
+"See this step in Github":https://github.com/ably/tutorials/commit/66d4a74b1af24ac32b0efebffa23924fb71addcb
+
+h2(#walking-time). Step 5 - Get location & find walking time
+
+Now we will get the location and update the walking time to the user selected station. 
+So for that we will need to use the Google API to get the walking time.
+Follow the steps "here":https://developers.google.com/maps/documentation/distance-matrix/get-api-key , to get the API key.
+Subscribe to the product Ably hub for the TFL London tube schedule "here":https://www.ably.io/hub/products/10 .
```

Comment:
Space before full stop.

Reply:
I think this sentence is a bit confusing currently, as it sounds like the product is Ably hub. I'd suggest something like `You can then subscribe to the "TFL Tube Schedule":https://www.ably.io/hub/products/10 via the "Ably Hub":https://www.ably.io/hub.`